packages/opds-web-client/src/authMiddleware.ts

```diff
@@ -9,7 +9,7 @@ import { AuthCallback, AuthProvider, AuthMethod, PathFor } from "./interfaces";
     See Redux Middleware docs:
     http://redux.js.org/docs/advanced/Middleware.html */
 
-export default (authPlugins: AuthPlugin[], pathFor: PathFor) => {
+export default (authPlugins: AuthPlugin[], pathFor: PathFor, test: boolean = false) => {
```

Comment:
@aslagle I don't currently see another way of solving this except by doing this technique which is used often in the server code. The tests run and the login form should work now.

Reply:
I think I figured it out. The resolve that's supposed to get called is on line 73, in the callback that's passed to `showAuthMiddleware`. But `showAuthMiddleware` is a stub, so you have to use `showAuthFormStub.callsArg(0);` to make it resolve.

Reply:
That makes sense. Thanks!